tests/gis_tests/geoapp/tests.py

```diff
@@ -448,6 +448,17 @@ def test_relate_lookup(self):
             self.assertEqual('Texas', Country.objects.get(mpoly__relate=(pnt2, intersects_mask)).name)
             self.assertEqual('Lawrence', City.objects.get(point__relate=(ks.poly, intersects_mask)).name)
 
+        # With a complex geometry expression
+        self.assertFalse(City.objects.exclude(point__relate=(functions.Union('point', 'point'), within_mask)))
+
+    def test_gis_lookups_with_complex_expressions(self):
+        multiple_arg_lookups = {'dwithin', 'relate'}  # These lookups are tested in other places.
+        lookups = connection.ops.gis_operators.keys() - multiple_arg_lookups
+        self.assertTrue(lookups, 'No lookups found')
+        for lookup in lookups:
+            with self.subTest(lookup):
+                City.objects.filter(**{'point__' + lookup: functions.Union('point', 'point')}).exists()
```

Comment:
That `Union` is aggregate.
Results differ among lookups, I don't think that it makes much sense to check them.